pkg/ocp/actuator.go

```diff
@@ -121,47 +110,113 @@ func (a *OCPActuator) GetCredentialsRootSecretLocation() types.NamespacedName {
+		return &actuatoriface.ActuatorError{
+			ErrReason: minterv1.CredentialsProvisionFailure,
+			Message:   fmt.Sprintf("%v: %v", msg, err),
+		}
+	}
+	return nil
+}
 
+func (a *OCPActuator) getCredentialsSecretData(ctx context.Context, logger log.FieldLogger) ([]byte, error) {
 	// get the secret of the kubevirt credentials
 	kubevirtCredentialsSecret := &corev1.Secret{}
 	if err := a.Client.Get(ctx, a.GetCredentialsRootSecretLocation(), kubevirtCredentialsSecret); err != nil {
 		msg := "unable to fetch root cloud cred secret"
 		logger.WithError(err).Error(msg)
-		return &actuatoriface.ActuatorError{
+		return nil, &actuatoriface.ActuatorError{
 			ErrReason: minterv1.CredentialsProvisionFailure,
 			Message:   fmt.Sprintf("%v: %v", msg, err),
 		}
 	}
 
-	// get the value from the secret
+	// get the secret data - the infra kubeconfig
 	infraClusterKubeconfig, ok := kubevirtCredentialsSecret.Data[KubevirtCredentialsSecretKey]
 	if !ok {
-		return errors.New("invalid mode")
+		return nil, errors.New("invalid mode")
+	}
+
+	logger.Debug("extracted kubevirt credentials")
+	return infraClusterKubeconfig, nil
+}
+
+func (a *OCPActuator) syncCredentialSecret(ctx context.Context, cr *minterv1.CredentialsRequest, kubevirtCredentialData *[]byte, existingSecret *corev1.Secret, logger log.FieldLogger) error{
+	sLog := logger.WithFields(log.Fields{
+		"targetSecret": fmt.Sprintf("%s/%s", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name),
+		"cr":           fmt.Sprintf("%s/%s", cr.Namespace, cr.Name),
+	})
+
+	if existingSecret == nil {
+		if kubevirtCredentialData == nil {
+			msg := "new access key secret needed but no key data provided"
+			sLog.Error(msg)
+			return &actuatoriface.ActuatorError{
+				ErrReason: minterv1.CredentialsProvisionFailure,
+				Message:   msg,
+			}
+		}
+
+		return a.createNewSecret(sLog, cr, kubevirtCredentialData, ctx)
+	}
+
+	return a.updateExistingSecret(sLog, existingSecret, cr, kubevirtCredentialData)
+}
+
+func (a *OCPActuator) updateExistingSecret(sLog *log.Entry, existingSecret *corev1.Secret, cr *minterv1.CredentialsRequest, kubevirtCredentialData *[]byte) error {
+	// Update the existing secret:
+	sLog.Debug("updating secret")
+	origSecret := existingSecret.DeepCopy()
+	if existingSecret.Annotations == nil {
+		existingSecret.Annotations = map[string]string{}
+	}
+	existingSecret.Annotations[minterv1.AnnotationCredentialsRequest] = fmt.Sprintf("%s/%s", cr.Namespace, cr.Name)
+	if kubevirtCredentialData != nil {
+		existingSecret.Data = map[string][]byte{
+			DataSecretKey: *kubevirtCredentialData,
+		}
+	}
+
+	if !reflect.DeepEqual(existingSecret, origSecret) {
+		sLog.Info("target secret has changed, updating")
+		err := a.Client.Update(context.TODO(), existingSecret)
+		if err != nil {
+			msg := "error updating secret"
+			sLog.WithError(err).Error(msg)
+			return &actuatoriface.ActuatorError{
+				ErrReason: minterv1.CredentialsProvisionFailure,
+				Message:   msg,
+			}
+		}
+	} else {
```

Comment:
else not needed because if do return

Reply:
The else is on line 194 (if !reflect.DeepEqual(existingSecret, origSecret)) 
So it needed